log4j-plugins/src/main/java/org/apache/logging/log4j/plugins/di/DefaultInstanceFactory.java

```diff
@@ -99,6 +104,9 @@ private DefaultInstanceFactory(
         this.bindings.put(Key.forClass(InstanceFactory.class), () -> this);
         this.bindings.put(Key.forClass(PropertyEnvironment.class), PropertiesUtil::getProperties);
         this.bindings.put(Key.forClass(ClassLoader.class), LoaderUtil::getClassLoader);
+        this.bindings.put(
+                Key.forClass(ConstraintValidatorFactory.class),
+                () -> new DefaultConstraintValidatorFactory(Integer::valueOf));
```

Comment:
I suppose this works for now, though we may be able to move most (or all) of these default bindings to a `ConfigurableInstanceFactoryPostProcessor` as I noted in a todo comment.